services/user.service.js

```diff
@@ -0,0 +1,39 @@
+const User = require("../schemas/user.schema");
+
+const createOne = async ({ name, email, password, salt }) => {
+  const newUser = await User.create({ name, email, password, salt });
+  await newUser.save();
+  return newUser;
+};
+
+const readOne = async (name) => {
+  return await User.findOne({ name });
+};
```

Comment:
Can 2 different users have the same name? In that case I think that you will return both. 
If I am correct is this intentional?

Reply:
No, 2 users can't have the same name. It will be enforced via the register methods in the user controller